homeassistant/components/aqara/binary_sensor.py

```diff
@@ -0,0 +1,221 @@
+"""Support for Aqara binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from aqara_iot import AqaraDeviceManager, AqaraPoint
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HomeAssistantAqaraData
+from .base import AqaraEntity, find_aqara_device_points_and_register
+from .const import AQARA_BATTERY_LOW_ENTITY_NEW, AQARA_DISCOVERY_NEW, DOMAIN
+
+
+@dataclass
+class AqaraBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Describes a Aqara binary sensor."""
+
+    # Value to consider binary sensor to be "on"
+    on_value: Any = True
+    on_icon: str = "mdi:restart"
+    off_icon: str = "mdi:restart"
+
+    def set_key(self, key) -> AqaraBinarySensorEntityDescription:
+        """Set key of binary Description."""
+        self.key = key
+        return self
+
+    def set_name(self, name) -> AqaraBinarySensorEntityDescription:
+        """Set name of binary Description."""
+        self.name = name
+        return self
```

Comment:
This method is unused and should not be needed. Entity descriptions are meant to be static

Reply:
removed